server/services/deviceWearerService.ts

```diff
@@ -1,40 +1,53 @@
 import RestClient from '../data/restClient'
 import logger from '../../logger'
-import { DeviceWearerResponse } from '../data_models/deviceWearer'
+import { ApiResponse } from '../utils/utils'
+import DeviceWearer from '../data_models/deviceWearer'
+import config from '../config'
 
 export default class DeviceWearerService {
-  private restClient: RestClient
-
-  constructor(injectedRestClient: RestClient) {
-    this.restClient = injectedRestClient
+  private restClient(token: string) {
+    return new RestClient('Data Platform API Client', config.apis.deviceWearer, token)
   }
 
-  async findOne(accessToken: string, deviceWearerId: string): Promise<DeviceWearerResponse> {
-    let result: DeviceWearerResponse
+  async findOne(accessToken: string, deviceWearerId: string): Promise<DeviceWearer> {
     try {
       logger.debug(`calling deviceWearerService.findOne`)
-      result = (await this.restClient.get({ path: `/device-wearers/v1/id/${deviceWearerId}` })) as DeviceWearerResponse
+      const response = await this.restClient(accessToken).get<ApiResponse<'deviceWearers', DeviceWearer[]>>({
+        path: `/device-wearers/v1/id/${deviceWearerId}`,
+      })
+      if (response.deviceWearers.length === 0) {
+        throw new Error(`No user found with ID ${deviceWearerId}`)
+      }
+      if (response.deviceWearers.length !== 1) {
+        throw new Error(`Duplicate users found with ID ${deviceWearerId}`)
+      }
+      return response.deviceWearers[0]
     } catch (e) {
-      logger.error({ err: e }, 'failed to fetch')
+      logger.error({ err: e }, `Failure in DeviceWearerService.findOne with deviceWearerId ${deviceWearerId}`)
       throw e
     }
-
-    return result
   }
 
-  async findMany(accessToken: string, searchTerm: string = null): Promise<DeviceWearerResponse> {
-    let result: DeviceWearerResponse
+  async findMany(accessToken: string, searchTerm: string = null): Promise<Array<DeviceWearer>> {
+    let result: Array<DeviceWearer>
     try {
       logger.debug(`calling deviceWearerService.findMany, with searchterm ${searchTerm}`)
       if (searchTerm) {
-        result = (await this.restClient.get({
+        const response = await this.restClient(accessToken).get<ApiResponse<'deviceWearers', DeviceWearer[]>>({
           path: `/device-wearers/v2/search/${searchTerm}`,
-        })) as DeviceWearerResponse
+        })
+        if (response.deviceWearers.length === 0) {
+          throw new Error(`No matching users found with search term ${searchTerm}`)
+        }
```

Comment:
I don't think this is an error

Reply:
It's not an API error, but we considered it was possibly a service error/controller error on the UI.
I'd be happy to make this not an error, but then we need to handle the empty response in the controller instead, because showing the UI page with just no content is a poor solution.